py_hamt/hamt.py

```
@@ -600,3 +638,70 @@ def ids(self):
+
+
+class TransformedHamt(HAMTOriginal):
+    """
+    A wrapper around the HAMT class that applies a transformation function
+    when setting or getting items, except for specific Zarr metadata keys.
+    """
+
+    ZARR_METADATA_KEYS = {".zarray", ".zgroup", ".zattrs", ".zmetadata"}
+    INDEX_KEY_PATTERN = re.compile(r".*/\d+$")  # Matches keys like lat/0, lon/0, time/0
+
+    def __init__(self, *args, transformer=None, encrypt_vars=[], **kwargs):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)